Observer/SalesOrderShipmentAfter.php

```diff
@@ -131,17 +152,27 @@ public function execute(\Magento\Framework\Event\Observer $observer)
             $this->createInvoice($order, $shipment);
         }
 
-        if (($payment->getMethodInstance()->getCode() == 'buckaroo_magento2_afterpay20')
+        if (($paymentMethodCode == 'buckaroo_magento2_afterpay20')
             && $this->afterpayConfig->getCreateInvoiceAfterShipment()
-            && ($payment->getMethodInstance()->getConfigPaymentAction() == 'authorize')
+            && ($paymentMethod->getConfigPaymentAction() == 'authorize')
         ) {
             $this->gateway->setMode(
                 $this->helper->getMode('buckaroo_magento2_afterpay20')
             );
             $this->createInvoice($order, $shipment, true);
         }
+
+        if (strpos($paymentMethodCode, 'buckaroo_magento2') !== false
+            && $this->configAccount->getInvoiceHandling() == InvoiceHandlingOptions::SHIPMENT) {
+            if ($paymentMethod->getConfigPaymentAction() == 'authorize') {
+                $this->createInvoice($order, $shipment, true);
+            } else {
+                $this->createInvoice->createInvoiceGeneralSetting($order);
```

Comment:
having a method and a property name the same way looks confusing to me

Reply:
I will change the property name to  createInvoiceService